mesonbuild/interpreter/interpreterobjects.py

```diff
@@ -974,6 +974,10 @@ def process_method(self,
             FeatureNew.single_use(
                 f'Calling generator.process with CustomTaget or Index of CustomTarget.',
                 '0.57.0', self.interpreter.subproject)
+        if any((isinstance(a, (build.Target)) and not isinstance(a, (build.CustomTarget))) for a in args[0]):
```

Comment:
I think this should just be is BuildTarget, which is IIRC every target except CustomTarget

Reply:
No there also is RunTarget which is also a Target, so using BuildTarget is no option.

Reply:
I don't think generators can depend on RunTargets. I think RunTargets are only allowed to depend on each other.

Reply:
Yeah, run targets are conceptually "do something" like upload a generated file somewhere. Build targets are "create something with a specific output name" which is what this is for, since said output is meant to be used as input to the generator.